app/models/application.rb

```diff
@@ -35,9 +35,25 @@ class Application < ApplicationRecord
   scope :with_targeted_delivery_funding_eligibility, -> { where(targeted_delivery_funding_eligibility: true) }
 
   validate :schedule_cohort_matches
+  # TODO: add constraints into the DB after separation
+  validates :ecf_id,
+            presence: { message: "Enter an ECF ID" },
+            uniqueness: {
+              case_sensitive: false,
+              message: "ECF ID must be unique",
+            }, if: -> { Feature.ecf_api_disabled? }
 
   after_commit :touch_user_if_changed
 
+  # TODO: remove this and add default: "gen_random_uuid()" in the DB after separation
+  before_validation(on: :create) do
```

Comment:
Could we not do this in a migration to set the default UUID as we do in some of the other tables?

```
add_column :statements, :ecf_id, :uuid, default: "gen_random_uuid()", null: false
```

This way if the UUID isn't set by ECF it will be auto-generated? it might be worth adding a couple of tests to ensure if we preset a uuid it doesn't change on creation if already populated but is set if `nil`?

Reply:
Well yeah this is the idea, that's why I added the TODO message to be done after separation.. but we can't do it now as there's logic behind it in `ApplicationSubmissionJob` where it fetches new applications still without ecf_id to be synced with ECF.. same for user really.

Reply:
I suppose the key thing here is that it generates `ecf_id` only when the feature flag is on.

Reply:
Should it only be on `create`?
I'm thinking of scenario when Application that's pending in NPQ, they haven't been accepted yet, so they won't be `ecf_id`. They later get accepted (after migration) and won't ever get `ecf_id` generated.

Maybe set it to all `before_validation` and when application is accepted?

Reply:
Tks @mooktakim. I've updated the callback to all `before_validation` which would make no harm and will cover when an application is accepted as well. So we'll have an `ecf_id` set if none after the feature flag is on whenever an application is created or updated.